content/en/docs/zero-code/python/troubleshooting.md

```diff
@@ -91,6 +92,105 @@ if __name__ == "__main__":
     app.run(port=8082, debug=True, use_reloader=False)
 ```
 
+### Pre-fork server issues
+
+A pre-fork server, such as Gunicorn with multiple workers, could be run like
+this:
+
+```sh
+gunicorn myapp.main:app --workers 4
+```
+
+However, specifying more than one `--workers` may break the generation of
+metrics when auto-instrumentation is applied. This is because forking, the
+creation of worker/child processes, creates inconsistencies between each child
+in the background threads and locks assumed by key OpenTelemetry SDK components.
+Specifically, the PeriodicExportingMetricReader spawns its own thread to
+periodically flush data to the exporter. See also issues
+[#2767](https://github.com/open-telemetry/opentelemetry-python/issues/2767) and
+[#3307](https://github.com/open-telemetry/opentelemetry-python/issues/3307#issuecomment-1579101152).
+After forking, each child seeks a thread object in memory that is not actually
+run, and any original locks may not unlock for each child. See also forks and
+deadlocks described in [Python issue 6721](https://bugs.python.org/issue6721).
+
+#### Workarounds
+
+There are some workarounds for pre-fork servers with OpenTelemetry. The
+following table summarizes the current support of signal export by different
+auto-instrumented web server gateway stacks that have been pre-forked with
+multiple workers. See below for more details and options:
+
+| Stack with multiple workers | Traces | Metrics | Logs |
+| --------------------------- | ------ | ------- | ---- |
+| Uvicorn                     | x      |         | x    |
+| Gunicorn                    | x      |         | x    |
+| Gunicorn + UvicornWorker    | x      | x       | x    |
+
+##### Deploy with Gunicorn and UvicornWorker
+
+To auto-instrument a server with multiple workers, it is recommended to deploy
+using Gunicorn with `uvicorn.workers.UvicornWorker` if it is an Asynchronous
+Server Gateway Interface (ASGI) app (FastAPI, Starlette, etc). The UvicornWorker
+class is specifically designed to handle forks with preservation of background
+processes and threads. For example:
+
+```sh
+opentelemetry-instrument gunicorn \
+  --workers 4 \
+  --worker-class uvicorn.workers.UvicornWorker \
+  --bind 0.0.0.0:8000 \
+  myapp.main:app
+```
+
+##### Use manual or programmatic instrumentation
+
+Initialize OpenTelemetry inside the worker process with
+[manual or programmatic instrumentation](/docs/zero-code/python/example/) after
+the server fork, instead of with auto-instrumentation. For example:
```

Comment:
nit this reads a little weird beacuse it says "instead of of auto-instrumentation" then the example shows `from opentelemetry.instrumentation.auto_instrumentation`. IMO this is still auto/zero code instrumentation in spirit. So maybe something like "set up manual instrumentation or zero-code instrumentation after the server fork".

Reply:
In the opentelemetry-instrumentation README it's called "Programmatic Auto-instrumentation"

Reply:
Thank you both for pointing this out; yeah it is odd. What about this instead, including updated link?:

```
##### Use programmatic auto-instrumentation

Initialize OpenTelemetry inside the worker process with
[programmatic auto-instrumentation](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/opentelemetry-instrumentation/README.rst#programmatic-auto-instrumentation) after
the server fork, instead of with `opentelemetry-instrument`. For example:
```

Reply:
Changed in [5f42c11](https://github.com/open-telemetry/opentelemetry.io/pull/7081/commits/5f42c115dbb5b42259cefad0b797f8e4fbc3ec46)